### ISSUE TITLE
Exception when using Measure and undoing operation

### ISSUE THREAD

**Issue by mjack3k** [FreeCAD/FreeCAD#18302]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

Application ends up in unusable state when undoing operation when Measure was active. 

Initial part:
![obraz](https://github.com/user-attachments/assets/b64ec1f5-84df-47c8-a244-07b5bcbcc965)

After adding chamfer and measuring the selected edge:
![obraz](https://github.com/user-attachments/assets/a3d6ab2e-3037-409d-8583-68b2fd66f45c)

While Measure is active, Ctrl-Z to undo the chamfer:
![obraz](https://github.com/user-attachments/assets/d1f80a8d-7d9c-41e3-8a73-40f17052a401)

When trying to close the Measure:
![obraz](https://github.com/user-attachments/assets/f1e5d3b9-7190-47d7-9335-6e810cdcfab6)

### Full version info

```shell
Basically:
- have a circular hole in your part
- create chamfer on the edge on that hole
- use measure tool on that edge
- use Ctrl-Z to undo the chamfer operation (while Measure is still active)
- tada! it is now stuck in Measure, can't close it, have to restart FreeCAD

It is happening in both Part and Part Design, also for simple straight edges (chamfer, measure, undo, stuck). 
It is possible to close the file and create new one, but the program is already borked. Only restart helps.
```

### Subproject(s) affected?

None

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by luzpaz** [comment]
Full About info please. (click that copy to clipboard button and paste into ticket)

**Comment by luzpaz** [comment]
Can you also please attach an example file (zip it up and drag/drop it into github textbox)

**Comment by mjack3k** [comment]
Example file:
[test1.zip](https://github.com/user-attachments/files/18013901/test1.zip)

It is really pretty minimal, to trigger the problem an operation like chamfer needs to be applied to any edge, then it has to be measured, then undo the operation while still in measure mode.

**Comment by luzpaz** [comment]
> Full About info please. (click that copy to clipboard button and paste into ticket)

Thanks, can you also add this?

**Comment by mjack3k** [comment]
OS: Void Linux (LXQt/lxqt/xcb)
Architecture: x86_64
Version: 1.0.0.1_voidlinux
Build type: Release
Python 3.12.7, Qt 6.7.2, Coin 4.0.0, Vtk 9.3.1, OCC 7.7.1
Locale: Polish/Poland (pl_PL)
Stylesheet/Theme/QtStyle: FreeCAD Dark.qss/FreeCAD Dark/
Installed mods:

    Gridfinity 0.6.0

**Comment by luzpaz** [comment]
@mjack3k can you test on 1.1.0dev ?

**Comment by mjack3k** [comment]
@luzpaz I will try. Life has been busy lately, maybe this weekend. 

Update: @luzpaz sorry for dumb question, but where do I find this binary?

**Comment by luzpaz** [comment]
@mjack3k apologies, didn't see the notification you responded.  
https://github.com/FreeCAD/FreeCAD/releases

**Comment by FEA-eng** [comment]
It looks quite serious. I get "Access violation" when using Ctrl+Z with the Measure task panel open for any case (even just Part Cube) and any measurement.

```
11:37:43  Unhandled Base::Exception caught in GUIApplication::notify.
The error message is: Access violation
```

There's also a new report on the forum: https://forum.freecad.org/viewtopic.php?t=101046

```
OS: Windows 11 build 26200
Architecture: x86_64
Version: 1.1.0dev.20251104 (Git shallow)
Build date: 2025/11/04 19:07:44
Build type: Release
Branch: main
Hash: af014a1bb44264d749ba9522917f0d1ac76c4bbf
Python 3.11.14, Qt 6.8.3, Coin 4.0.3, Vtk 9.3.1, boost 1_86, Eigen3 3.4.0, PySide 6.8.3
shiboken 6.8.3, xerces-c 3.3.0, IfcOpenShell 0.8.2, OCC 7.8.1
Locale: Polish/Poland (pl_PL)
Navigation Style/Orbit Style/Rotation Mode: Blender/Rounded Arcball/Window center
Stylesheet/Theme/QtStyle: FreeCAD.qss/FreeCAD Light/
Logical DPI/Physical DPI/Pixel Ratio: 96/81.5973/1
```

**Comment by luzpaz** [comment]
@maxwxyz looks like a blocker since it can catastrophically disrupt workflow (especially as an undo/redo bug)

**Comment by pjcreath** [comment]
This was coincidentally resolved by #25129 and should be working in [today's weekly build](https://github.com/FreeCAD/FreeCAD/releases/tag/weekly-2025.11.12).

**Comment by mjack3k** [comment]
@pjcreath those are some great news. I will try to check it out tonight.

**Comment by luzpaz** [comment]
Closed. Please re-open if you can reproduce.
